protos/sigstore_trustroot.proto

```diff
@@ -117,3 +117,43 @@ message TrustedRoot {
         // A set of trusted timestamping authorities.
         repeated CertificateAuthority timestamp_authorities = 5;
 }
+
+
+// SigningConfig represents the trusted entities/state needed by Sigstore
+// signing. In particular, it primarily contains service URLs that a Sigstore
+// signer may need to connect to for the online aspects of signing.
+message SigningConfig {
+        // A URL to a Fulcio-compatible CA, capable of receiving
+        // Certificate Signing Requests (CSRs) and responding with
+        // issued certificates.
+        //
+        // This URL **MUST** be the "base" URL for the CA, which clients
+        // should construct an appropriate CSR endpoint on top of.
+        // For example, if `fulcio_url` is `https://example.com/ca`, then
+        // the client **MAY** construct the CSR endpoint as
+        // `https://example.com/ca/api/v2/signingCert`.
+        string fulcio_url = 1;
+
+        // A URL to an OpenID Connect identity provider.
+        //
+        // This URL **MUST** be the "base" URL for the OIDC IdP, which clients
+        // should perform well-known OpenID Connect discovery against.
+        string oidc_url = 2;
+
+        // A URL to a Rekor-compatible transparency log.
+        //
+        // This URL **MUST** be the "base" URL for the transparency log,
+        // which clients should construct appropriate API endpoints on top of.
+        string rekor_url = 3;
```

Comment:
do we need a timestamp authority url?

Reply:
Ah, yep, good catch!

sigstore-python doesn't currently do any TSA stuff, so to confirm: would this URL be expected to be the base endpoint for receiving TSRs, or should the client know to append/build an endpoint on top of it?

Reply:
Hmm, I don't think that's standard 😅 -- I believe RFC 3161 only stipulates an HTTP POST to a pre-defined endpoint of a DER-encoded TSR. I'll look into it.

Reply:
Yeah, RFC 3161 is pretty badly underspecified here: it says HTTP, but doesn't specify an HTTP method or a particular endpoint. It _does_ however stipulate a content-type and that the payload is just the bare DER for the TSR, which isn't true in this case. So I _think_ we want to stipulate that the URL here behaves like <https://github.com/sigstore/timestamp-authority>.

CC @haydentherapper for opinions here 🙂

Reply:
Yeah, I looked up other TSAs and it's all a bit different. I don't know if @bdehamer has an opinion. Like should the TSA endpoint just be a full address?

Reply:
Yeah, I think we should probably just stipulate that this is a Sigstore-shaped TSA. There isn't enough normative standards language to really enforce anything here and it only matters for signing (since verifying is universal across TSAs).

Reply:
Would any rfc3161 timestamp satisfy the requirements of sigstore?

Reply:
Added `tsa_url` with docs explaining that it should behave like the Sigstore TSA.

Reply:
> Would any rfc3161 timestamp satisfy the requirements of sigstore?

On the verifying side yeah, but on the signing side we need to know *how* to submit the TSR to the TSA. That part isn't well-specified, hence the need for concrete behavior here 🙂

Reply:
Sorry if I was unclear, I think we should do the opposite and require the full path. The RFC specifies the expected payload and content type. You're right that it doesn't specify HTTP method, but in practice, all of the TSAs we tested used POST. The [golang implementation](https://github.com/sigstore/cosign/blob/main/internal/pkg/cosign/tsa/client/client.go#L120 ), with the RFC-compliant TSR and content header, worked for more than just the sigstore TSA as long as the full API path was provided.

Reply:
Nope, I just somehow missed your message 😅 -- I'm going to blame that on conference Wi-Fi. 

I totally missed that the Sigstore TSA accepts the `application/timestamp-query` content-type in addition to JSON (which is what sigstore-js seems to use). Given that, fully agreed on passing the full URL, and I'll update for that 🙂

Reply:
Updated, PTAL!

Reply:
Ah yes, we added JSON later on (https://github.com/sigstore/timestamp-authority/pull/247) for clients that didn't want to deal with DER, but the TSA impl is fully RFC3161 compliant.